### PR TITLE
automap overlay and rotate modes (based on Crispy Doom)

### PR DESCRIPTION
**Pull request by rfomin**

### CODE REVIEW

File: Source/d_main.c [hunk 1]
```diff
@@ -279,7 +279,7 @@ void D_Display (void)
     }
 
   // see if the border needs to be updated to the screen
-  if (gamestate == GS_LEVEL && !automapactive && scaledviewwidth != 320)
+  if (gamestate == GS_LEVEL && (!automapactive || automapoverlay) && scaledviewwidth != SCREENWIDTH)
```
Comment (fabiangreffrath) [inline Source/d_main.c:282]: Not sure about the `320` -> `SCREENWIDTH` change. :thinking:

File: Source/m_menu.c [hunk 2]
```diff
@@ -2591,12 +2591,13 @@ static int G_GotoNextLevel(void)
 #define X_BUTTON 301
 #define Y_BUTTON   3
 
-// Definitions of the (in this case) four key binding screens.
+// Definitions of the (in this case) five key binding screens.
```
Comment (fabiangreffrath) [inline Source/m_menu.c:2594]: OMG, we have done it! :grin:

File: Source/st_lib.c [hunk 3]
```diff
@@ -221,7 +221,7 @@ void STlib_updatePercent
   char *outrng,             //jff 2/16/98 add color translation to digit output
   int refresh )
 {
-  if (refresh || *per->n.on) // killough 2/21/98: fix percents not updated;
+  if (/*refresh || */*per->n.on) // killough 2/21/98: fix percents not updated;
```
Comment (fabiangreffrath) [inline Source/st_lib.c:224]: This looks dirty to me.

**Review by fabiangreffrath — CHANGES_REQUESTED** [review]

File: Source/hu_stuff.c [hunk 4]
```diff
@@ -753,13 +753,22 @@ void HU_Drawer(void)
 
   plr = &players[displayplayer];         // killough 3/7/98
   // draw the automap widgets if automap is displayed
-  if (automapactive)
+  if (automapactive && !(hud_distributed && automapoverlay))
     {
       fixed_t x,y,z;   // killough 10/98:
       void AM_Coordinates(const mobj_t *, fixed_t *, fixed_t *, fixed_t *);
 
+      if (!(hud_displayed && automapoverlay))
```
Comment (fabiangreffrath) [inline Source/hu_stuff.c:761]: Could this somehow get moved into `HU_Ticker()` instead?

File: Source/st_lib.c (hunk 3 above)
Comment (rfomin) [reply Source/st_lib.c:224]: I'm not sure about that. What was the original bug fixed here? If left unchanged, percent symbols are always displayed on-screen in overlay mode.

File: Source/m_menu.c (hunk 2 above)
Comment (rfomin) [reply Source/m_menu.c:2594]: BTW, I think we could reorganize the automap menu a bit like Pr +. Maybe add a transparent overlay mode option.

File: Source/st_lib.c (hunk 3 above)
Comment (fabiangreffrath) [reply Source/st_lib.c:224]: I guess this has to do with the percent signs not updating when switched between gray and colored.

File: Source/m_menu.c (hunk 2 above)
Comment (fabiangreffrath) [reply Source/m_menu.c:2594]: Yes, okay.

File: Source/d_main.c (hunk 1 above)
Comment (rfomin) [reply Source/d_main.c]: I revert it back.

File: Source/st_lib.c (hunk 3 above)
Comment (rfomin) [reply Source/st_lib.c]: I change it to `if (*per->n.on && refresh)`, seems to work.

File: Source/hu_stuff.c (hunk 4 above)
Comment (rfomin) [reply Source/hu_stuff.c:761]: Don't know about this.

File: Source/m_menu.c (hunk 2 above)
Comment (rfomin) [reply Source/m_menu.c:2594]: I think the automap menu and transparent mode will be in separate PR. This PR is already changing a lot.

File: Source/hu_stuff.c [hunk 5]
```diff
@@ -831,7 +840,7 @@ void HU_Drawer(void)
      hud_active>0 &&                  // hud optioned on
      hud_displayed &&                 // hud on from fullscreen key
      scaledviewheight==SCREENHEIGHT &&// fullscreen mode is active
-     !automapactive                   // automap is not active
+     ((automapactive && automapoverlay) || !automapactive)
```
Comment (fabiangreffrath) [inline Source/hu_stuff.c:843]: `(!automapactive || automapoverlay)`?

File: Source/hu_stuff.c (hunk 4 above)
Comment (fabiangreffrath) [reply Source/hu_stuff.c:761]: There is a block in `HU_Ticker()` where the level stats and level time widgets are filled. Could you move this block which adjusts the vertical position of the map title widget to that other block?

File: Source/st_lib.c [hunk 6]
```diff
@@ -221,7 +221,7 @@ void STlib_updatePercent
   char *outrng,             //jff 2/16/98 add color translation to digit output
   int refresh )
 {
-  if (refresh || *per->n.on) // killough 2/21/98: fix percents not updated;
+  if (*per->n.on && refresh) // killough 2/21/98: fix percents not updated;
```
Comment (fabiangreffrath) [inline Source/st_lib.c:224]: Sorry, but this is how it was in Boom 2.01, i.e. before it got "fixed":
https://github.com/Doom-Utils/historic-ports/blame/ce8bbd699f29452edf7d02b912b49146a80144ed/st_lib.c#L184

Looking at `STlib_updateNum()` and related functions, it seems that the `refresh` value is outright ignored there. So, I am all for rolling back to your initial approach of removing the check for `refresh` - but please add a comment.

**Review by fabiangreffrath — CHANGES_REQUESTED** [review]

File: Source/st_lib.c (hunk 6 above)
Comment (rfomin) [reply Source/st_lib.c]: Done in https://github.com/fabiangreffrath/woof/pull/216/commits/6446fe0266ada9f3c536925118f6c5d63273943a

**Comment by fabiangreffrath** [comment]
Thank you!

File: Source/m_menu.c (hunk 2 above)
Comment (fabiangreffrath) [reply Source/m_menu.c:2594]: Okay, so we still need to rearrange the menu here.

Comment (rfomin) [reply Source/m_menu.c:2594]: I made translucent overlap mode, but it uses `main_tranmap`. Maybe we should make a separate tranmap with a filter percentage option but I am not sure if it's worth it.

Comment (fabiangreffrath) [reply Source/m_menu.c:2594]: Hm, could you show a screenshot with the default translucency settings?

Comment (rfomin) [reply Source/m_menu.c:2594]: 66%:
doom01
translucency off:
doom00
33%:
doom02

So, I guess we need a separate tranmap or scrap the idea.

Comment (fabiangreffrath) [reply Source/m_menu.c:2594]: Okay, thank you! The difference is indeed only barely visible, at least in still images. I wouldn't say "scrap the idea" but maybe postpone for later. ;)

Comment (rfomin) [reply Source/m_menu.c:2594]: > postpone for later

I agree this feature is very minor. To be honest, I didn't know Crispy and Pr+ had these features all these years. :smile:

**Pull request merged**